src/v/kafka/server/protocol.cc

```diff
@@ -60,7 +61,33 @@ protocol::protocol(
       config::shard_local_cfg().enable_idempotence.value())
   , _credentials(credentials)
   , _authorizer(authorizer)
-  , _security_frontend(sec_fe) {}
+  , _security_frontend(sec_fe) {
+    /*
+     * start the queue depth control algorithm. on each timer invocation we
+     * advance the latency tracker and optionally update the queue depth.
+     */
+    auto depth_update_freq
+      = config::shard_local_cfg().kafka_qdc_depth_update_ms();
+
+    _qdc.timer.set_callback([this, depth_update_freq] {
+        auto sample = _qdc.ema.tick();
+        auto now = ss::lowres_clock::now();
+        if ((now - _qdc.last_update) < depth_update_freq) {
+            return;
+        }
+        _qdc.last_update = now;
+        _qdc.qdc.update(sample);
+        vlog(
+          klog.debug,
+          "Updating queue depth to {} at latency {}",
+          _qdc.qdc.depth(),
+          sample);
+    });
+
+    _qdc.timer.arm_periodic(
```

Comment:
pass in as a struct of values rather than relying on the core-local config. global state is hard to test for/debug

Reply:
specially as we start to do more in process multi redpanda nodes with different timeouts.

Reply:
this PR is not a clean-up PR and this kafka server type this is embedded already takes config options from core local.

Reply:
ack